client.go

```diff
@@ -18,6 +19,13 @@ var (
 	apiUrlV2 = "https://api.aiven.io/v2"
 )
 
+var (
+	// errUnableToCreateAivenClient is a template error for when the client cannot be created.
+	errUnableToCreateAivenClient = func(err error) error {
```

Comment:
I wonder if it's necessary to split this error handling here :thinking: It doesn't seems to make the code less duplicated, nor easier to read. I would suggest we drop this, and just use `fmt.Errorf()`

Reply:
> I wonder if it's necessary to split this error handling here 🤔 It doesn't seems to make the code less duplicated, nor easier to read. I would suggest we drop this, and just use `fmt.Errorf()`

please see https://github.com/aiven/aiven-go-client/pull/182#discussion_r949119976